source/core_http_client.c

```diff
@@ -870,6 +870,14 @@ static int httpParserOnHeadersCompleteCallback( llhttp_t * pHttpParser )
 
     LogDebug( ( "Response parsing: Found the end of the headers." ) );
 
+    /* If there is HTTP_RESPONSE_DO_NOT_PARSE_BODY_FLAG opt-in we should stop
+     * parsing here. */
+    if( pResponse->respOptionFlags & HTTP_RESPONSE_DO_NOT_PARSE_BODY_FLAG )
```

Comment:
I would like to bring to your attention that, at this stage, we will receive the body along with the header section terminator \r\n\r\n. I have not included stripping in this section because it can be easily implemented in our library. However, if it aligns well with corehttp's functionality, we can modify the code to include stripping at this point.

Reply:
Yeah, would make sense for body to be set to after the terminator, since we'll need to have read the headers terminator anyways.

Reply:
@cryi would you mind making the change to consume the terminator?

Reply:
This is addressed in the commit I made. We do not need to explicitly skip the end marker as we can utilized the `error_pos` member of the parser which is updated whenever the parser is paused.